src/extension.ts

```diff
@@ -802,6 +803,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<any> {
       continue;
     }
   }
+  for await (const workspaceFolder of vscode.workspace.workspaceFolders) {
```

Comment:
Do we have to `await` and delay activation? Also, this needs to handle the case where `workspaceFolders` is `undefined`.

Reply:
Good catch about the possibility of it being undefined. Re await, I'd like to be sure that we've gathered the data before the FileSystemProvider starts being used. I'd expect that not to happen before activate has returned. So I'd prefer to leave the await in.